GVFS/GVFS.Service/RepoRegistry.cs

```diff
@@ -21,12 +21,26 @@ public class RepoRegistry
         private ITracer tracer;
         private PhysicalFileSystem fileSystem;
         private object repoLock = new object();
+        private IRepoMountProcess mountProcess;
+        private GVFSPlatform gvfsPlatform;
 
-        public RepoRegistry(ITracer tracer, PhysicalFileSystem fileSystem, string serviceDataLocation)
+        // The no-args Constructor is there only because it is needed by Moq
+        public RepoRegistry()

```

Comment:
Is this strictly necessary? If we're going to mock the methods we need, can't we call the below ctor with nulls?

Reply:
This is another reason to create an interface that `RepoRegistry` implements and then mock the interface.  Mocking with real objects seems to be a little less straight forward.

Reply:
Added a new interface for `RepoRegistry`. This is not needed anymore.